sw.js

```diff
@@ -0,0 +1,50 @@
+const CACHE_VERSION_STATIC = 'cache-version-s-1'
+const CACHE_VERSION_DYNAMIC = 'cache-version-d-1'
+
+const assetsURL = ['/index.html', '/favicon.svg']
+
+const networkFirst = async request => {
+  // создаем кэш для
+  const cache = await caches.open(CACHE_VERSION_DYNAMIC)
+
+  try {
+    // получаем файлы с сервера
+    const response = await fetch(request)
+
+    // сохраняем в кеш
+    await cache.put(request, response.clone())
+
+    return response
+  } catch (err) {
+    // достаем из кэша сохраненные данные, если они не подгрузились с сервера
+    return cache.match(request)
+  }
+}
+
+self.addEventListener('install', event => {
```

Comment:
Слово `self` лишнее. Оно и так по умолчанию подразумевается.

```suggestion
addEventListener('install', event => {
```